discovery/discovery_test.go

```diff
@@ -335,26 +309,126 @@ func TestNewDBOrchestratorPoolCache_TestURLs_Empty(t *testing.T) {
 	assert := assert.New(t)
 	require.Nil(err)
 
-	node, _ := core.NewLivepeerNode(nil, "", nil)
-	node.Database = dbh
-
 	addresses := []string{}
 	// Addresses is empty slice -> No orchestrators
 	orchestrators := StubOrchestrators(addresses)
-	// Contains empty orchestrator slice -> No registered orchs found on chain error
-	node.Eth = &eth.StubClient{Orchestrators: orchestrators}
-	errorLogsBefore := glog.Stats.Error.Lines()
-	dbOrch := NewDBOrchestratorPoolCache(node)
-	errorLogsAfter := glog.Stats.Error.Lines()
-	require.NotNil(dbOrch)
-	assert.Equal(0, dbOrch.Size())
-	assert.NotZero(t, errorLogsAfter-errorLogsBefore)
-	urls := dbOrch.GetURLs()
+	for _, o := range orchestrators {
+		dbh.UpdateOrch(ethOrchToDBOrch(o))
+	}
+	node := &core.LivepeerNode{
+		Database: dbh,
+		Eth: &eth.StubClient{
+			Orchestrators: orchestrators,
+		},
+		Sender: &pm.MockSender{},
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	pool, err := NewDBOrchestratorPoolCache(ctx, node, &stubRoundsManager{})
+	require.NoError(err)
+	assert.Equal(0, pool.Size())
+	urls := pool.GetURLs()
 	assert.Len(urls, 0)
 }
 
+func TestNewDBOrchestorPoolCache_PollOrchestratorInfo(t *testing.T) {
+	cachedOrchInfo := &net.OrchestratorInfo{
+		Transcoder: "transcoderFromTest",
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  999,
+			PixelsPerUnit: 1,
+		},
+	}
+	polledOrchInfo := &net.OrchestratorInfo{
+		Transcoder: "transcoderFromTest",
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  1,
+			PixelsPerUnit: 1,
+		},
+	}
+	returnInfo := cachedOrchInfo
+
+	var mu sync.Mutex
+	callCount := 0
+	first := true
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+		mu.Lock()
+		if first {
+			time.Sleep(100 * time.Millisecond)
+			first = false
+		}
+		mu.Unlock()
+		callCount++
+		return returnInfo, nil
+	}
+
+	dbh, dbraw, err := common.TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	require := require.New(t)
+	assert := assert.New(t)
+	require.Nil(err)
+
+	// adding orchestrators to DB
+	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
+	orchestrators := StubOrchestrators(addresses)
+
+	testOrchs := make([]orchTest, 0)
+	expPrice, _ := common.PriceToFixed(big.NewRat(999, 1))
+	for _, o := range orchestrators {
+		to := orchTest{
+			EthereumAddr:  o.Address.String(),
+			ServiceURI:    o.ServiceURI,
+			PricePerPixel: expPrice,
+		}
+		testOrchs = append(testOrchs, to)
+	}
+
+	sender := &pm.MockSender{}
+	node := &core.LivepeerNode{
+		Database: dbh,
+		Eth: &eth.StubClient{
+			Orchestrators: orchestrators,
+		},
+		Sender: sender,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sender.On("ValidateTicketParams", mock.Anything).Return(nil)
+
+	cacheRefreshInterval = 200 * time.Millisecond
```

Comment:
Let's make sure to reset this to the original value after the test finishes in case any other tests depend on the variable value.

Ex.

```
origCacheRefreshInterval = cacheRefreshInterval
cacheRefreshInterval = 200 * time.Millisecond
defer func() { cacheRefreshInterval = origCacheRefreshInterval }()
```

Reply:
fixed in e8acbd03